### PR TITLE
Added ftp user to camera class, improved keyring check at deserialization time.

### PR DESCRIPTION
**Pull request by stefanodellosa-personal**

### CODE REVIEW

File: wadas/ui/configure_ftp_cameras_dialog.py [hunk 1]
```diff
@@ -164,9 +164,11 @@ def accept_and_close(self):
                                         f"WADAS_FTP_camera_{camera.id}", ""
                                     )
                                     if credentials and (
-                                            credentials.username != cur_user
-                                            or credentials.password != cur_pass
+                                        credentials.username == cur_user
+                                        or credentials.password == cur_pass
```
Comment (cdimauro) [inline wadas/ui/configure_ftp_cameras_dialog.py:168]: I think that here you should only check for the current user and not for the password. The user could be the same, but the password might have changed -> update the keyring.

File: wadas/ui/mainwindow.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import os
 from datetime import timedelta
 from logging.handlers import RotatingFileHandler
+from operator import truediv
```
Comment (cdimauro) [inline wadas/ui/mainwindow.py:7]: Why this import? I don't find truediv used somewhere else on this PR.

File: wadas/ui/mainwindow.py [hunk 3]
```diff
@@ -287,6 +294,24 @@ def update_toolbar_status(self):
             self.isWindowModified() and bool(self.configuration_file_name)
         )
 
+    def ftp_camera_exists(self):
+        """Method that checks if at least one FTP camera exists in camera list."""
+        for camera in cameras:
+            if camera.type == Camera.CameraTypes.FTP_CAMERA:
+                return True
+        return False
```
Comment (cdimauro) [inline wadas/ui/mainwindow.py:302]: return any(camera.type == Camera.CameraTypes.FTP_CAMERA for camera in cameras)

File: wadas/ui/mainwindow.py [hunk 4]
```diff
@@ -287,6 +294,24 @@ def update_toolbar_status(self):
...
+    def email_notifier_exists(self):
+        """Method that checks if email notifier is configured."""
+
+        for notifier in Notifier.notifiers:
+            if (
+                Notifier.notifiers[notifier]
+                and Notifier.notifiers[notifier].type == Notifier.NotifierTypes.EMAIL
+            ):
+                return True
+        return False
```
Comment (cdimauro) [inline wadas/ui/mainwindow.py:313]: return any(Notifier.notifiers[notifier] and Notifier.notifiers[notifier].type == Notifier.NotifierTypes.EMAIL
                  for notifier in Notifier.notifiers)

File: wadas/ui/mainwindow.py (hunk 2 above)
Comment (stefanodellosa-personal) [reply wadas/ui/mainwindow.py:7]: it's a rebase

Comment (stefanodellosa-personal) [reply wadas/ui/mainwindow.py:7]: @hexfati this shoudl be coming from your latest changes, correct?

Comment (hexfati) [reply wadas/ui/mainwindow.py:7]: nope, never seen before now

**Review by hexfati — APPROVED** [review]

**Pull request merged**